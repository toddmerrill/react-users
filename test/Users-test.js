import React from 'react'
import renderer from 'react-test-renderer'
import Users from '../src/components/Users.js'
import mock from './user-mocks'

it('Users renders correctly', () => {
    const tree = renderer.create(
        <Users users={{users: mock.users}} currentUser={mock.user} addUser={mock.f}
               deleteUser={mock.f} setCurrentUser={mock.f} persistUser={mock.f}
               updateUser={mock.f} onClick={mock.f}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
