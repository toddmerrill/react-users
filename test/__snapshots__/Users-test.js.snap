// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders correctly 1`] = `
<div>
  <h1
    className="title"
  >
    <img
      src="test-file-stub"
    />
     react users
  </h1>
  <div
    className="container"
  >
    <div
      className="leftList"
    >
      <div
        className="listHeader"
      >
        <h2
          className="oneLine"
        >
          Users
        </h2>
        <span
          className="rightButtons"
        >
          <a
            className="nohover addButton"
            onClick={[Function]}
          >
            Add User
          </a>
          <a
            className="nohover deleteButton"
            onClick={[Function]}
          >
            Delete User
          </a>
        </span>
      </div>
      <div
        className="listItems"
      >
        <div
          className="listrow current-row"
          onClick={[Function]}
        >
          <img
            src="test-file-stub"
          />
          test-first
           
          test-last
        </div>
      </div>
    </div>
    <div
      className="rightList"
    >
      <h2>
        Personal Information
      </h2>
      <form
        className="form-horizontal"
      >
        <div
          className="form-group"
        >
          <label
            className="controlLabel"
            htmlFor="inputfirstName"
          >
            First Name
          </label>
          <div
            className="controls"
          >
            <input
              autoFocus={true}
              className="form-control"
              id="inputfirstName"
              name="firstName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="First Name"
              tabIndex={1}
              type="text"
              value="test-first"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className="controlLabel"
            htmlFor="inputlastName"
          >
            Last Name
          </label>
          <div
            className="controls"
          >
            <input
              autoFocus={undefined}
              className="form-control"
              id="inputlastName"
              name="lastName"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Last Name"
              tabIndex={2}
              type="text"
              value="test-last"
            />
          </div>
        </div>
        <div
          className="form-group"
        >
          <label
            className="controlLabel"
            htmlFor="inputage"
          >
            Age
          </label>
          <div
            className="controls"
          >
            <input
              autoFocus={undefined}
              className="form-control"
              id="inputage"
              name="age"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder="Age"
              tabIndex={3}
              type="text"
              value="987"
            />
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
